out_writeapi.go

```diff
@@ -545,11 +563,17 @@ func FLBPluginFlushCtx(ctx, data unsafe.Pointer, length C.int, tag *C.char) int
 		log.Printf("Finding configuration for output instance with id: %d failed in FLBPluginFlushCtx", id)
 		return output.FLB_ERROR
 	}
+	// Holds stream slice for ease of use
+	streamSlice := *config.managedStreamSlice
```

Comment:
Do we need a mutex to control access to the stream slice? When will the slice potentially get modified?

Reply:
The slice gets modified when we add a new stream to the slice. When I implement this change, I will be adding a mutex around the insertion of the new stream into the slice!

Reply:
The slice also gets modified in init, but we only append one stream to it, so putting a mutex there will be unnecessary.

Reply:
Note that we will need to protect access to the slice with a mutex anywhere it is accessed, whether for read or for write.